examples/schedulers/static_thread_pool.hpp

```diff
@@ -88,11 +88,229 @@ namespace example {
+            shared_state& state = self.shared_state_;
+            tag((Receiver&&)state.receiver_, (As&&)as...);
+          }
+
+          friend auto tag_invoke(std::execution::get_env_t, const bulk_receiver& self)
+            -> std::execution::env_of_t<Receiver> {
+            return std::execution::get_env(self.shared_state_.receiver_);
+          }
+        };
+
+      template <class SenderId, class ReceiverId, std::integral Shape, class Fun>
+        struct bulk_op_state {
+          using Sender = std::__t<SenderId>;
+          using Receiver = std::__t<ReceiverId>;
+
+          using bulk_receiver = bulk_receiver<SenderId, ReceiverId, Shape, Fun>;
+          using shared_state = bulk_shared_state<SenderId, ReceiverId, Shape, Fun>;
+          using inner_op_state = std::execution::connect_result_t<Sender, bulk_receiver>;
+
+          shared_state shared_state_;
+
+          inner_op_state inner_op;
+
+          friend void tag_invoke(std::execution::start_t, bulk_op_state& op) noexcept {
+            std::execution::start(op.inner_op);
+          }
+
+          bulk_op_state(static_thread_pool &pool, Shape shape, Fun fn, Sender sender, Receiver receiver)
+            : shared_state_(pool, (Receiver&&)receiver, shape, fn)
+            , inner_op{std::execution::connect((Sender&&)sender, bulk_receiver{shared_state_})} {
+          }
+        };
+
+      template <class SenderId, std::integral Shape, class FunId>
+        struct bulk_sender {
+          using Sender = std::__t<SenderId>;
+          using Fun = std::__t<FunId>;
+
+          static_thread_pool& pool_;
+          Sender sndr_;
+          Shape shape_;
+          Fun fun_;
+
+          template <class Fun, class Sender, class Env>
+            using with_error_invoke_t =
+              std::__if_c<
+                std::__v<std::execution::__value_types_of_t<
+                  Sender,
+                  Env,
+                  std::__mbind_front_q<std::execution::__non_throwing_, Fun, Shape>,
+                  std::__q<std::__mand>>>,
+                std::execution::completion_signatures<>,
+                std::execution::__with_exception_ptr>;
+
+          template <class Self, class Env>
+            using completion_signatures =
+              std::execution::__make_completion_signatures<
+                std::__member_t<Self, Sender>,
+                Env,
+                with_error_invoke_t<Fun, std::__member_t<Self, Sender>, Env>>;
+
+          template <class Receiver>
+            using bulk_op_state_t = bulk_op_state<SenderId, std::__x<std::remove_cvref_t<Receiver>>, Shape, Fun>;
+
+          template <std::__decays_to<bulk_sender> Self, std::execution::receiver Receiver>
+          friend bulk_op_state_t<Receiver> tag_invoke(std::execution::connect_t, Self&& self, Receiver&& rcvr)
+            noexcept(std::is_nothrow_constructible_v<bulk_op_state_t<Receiver>, static_thread_pool&, Shape, Fun, Sender, Receiver>) {
+            return bulk_op_state_t<Receiver>{
```

Comment:
This isn't quite right. First, it's not constrained. Second, although you take the `bulk_sender` by forwarding reference, you aren't forwarding the `sndr_` member into the operation state, and the operation state isn't taking the cv- and ref-qualifiers of the sender into consideration.